prow/pubsub/subscriber/subscriber.go

```diff
@@ -154,16 +165,65 @@ func (peh *periodicJobHandler) getProwJobSpec(cfg *config.Config, pe ProwJobEven
 }
 
 // presubmitJobHandler implements jobHandler
-type presubmitJobHandler struct{}
+type presubmitJobHandler struct {
+	GitClient git.ClientFactory
+}
+
+func (prh *presubmitJobHandler) getProwJobSpec(cfg prowCfgClient, pe ProwJobEvent) (*v1.ProwJobSpec, map[string]string, error) {
+	// presubmit jobs require Refs and Refs.Pulls to be set
+	refs := pe.Refs
+	if refs == nil {
+		return nil, nil, errors.New("Refs must be supplied")
+	}
+	if len(refs.Pulls) == 0 {
+		return nil, nil, errors.New("at least 1 Pulls is required")
+	}
+
+	var presubmitJob *config.Presubmit
+	org, repo, branch := refs.Org, refs.Repo, refs.BaseRef
+	orgRepo := org + "/" + repo
+	baseSHAGetter := func() (string, error) {
+		return refs.BaseSHA, nil
+	}
+	var headSHAGetters []func() (string, error)
+	for _, pull := range refs.Pulls {
+		headSHAGetters = append(headSHAGetters, func() (string, error) {
+			return pull.SHA, nil
```

Comment:
This captures the local variable `pull` which changes each iteration of the loop. We should create a copy of `pull` inside the `for` loop and reference that within the anonymous function so that each function correctly references a different pull.

Reply:
good point, done